grails-app/migrations/views/inventory-counts.sql

```diff
@@ -88,23 +64,25 @@ CREATE OR REPLACE VIEW inventory_counts AS
     -- so filter it out
     baseline_without_adjustment AS (
         SELECT
-            CRC32(CONCAT(t.id, ii.product_id, facility.id)) as id,
-            t.id AS transaction_id,
+            CRC32(CONCAT(te.transaction_id, ii.product_id, facility.id)) as id,
+            te.transaction_id AS transaction_id,
             ii.product_id AS product_id,
             facility.id AS facility_id,
             t.transaction_date AS date_recorded,
             'BASELINE' as inventory_count_type_code,
             -- Alone baseline doesn't have any associated transaction (like baseline + adjustment), so hardcode it to NULL
             NULL as associated_transaction_id
-        FROM transaction t
-        JOIN transaction_entry te ON te.transaction_id = t.id
+        FROM transaction_entry te
+        JOIN transaction t ON t.id = te.transaction_id
         JOIN inventory_item ii ON ii.id = te.inventory_item_id
         JOIN location facility ON facility.inventory_id = t.inventory_id
+        LEFT JOIN transaction adjustment ON TIMESTAMPDIFF(SECOND, t.transaction_date, adjustment.transaction_date) = 1
         WHERE t.transaction_type_id = '12' -- baseline inventory transaction
-          AND NOT EXISTS (
+        AND NOT EXISTS (
               SELECT 1
               FROM baseline_adjustment_matches bam
               WHERE bam.baseline_id = t.id
+              AND bam.product_id = ii.product_id
```

Comment:
This is also a huge improvement and also a fix of a bug - when we do the inventory import, we actually create a baseline and adjustment for all items in the file, which means we can have multiple matches for one product (as X baselines contain this product) where X is amount of distinct items in the file, so by looking at the `baseline_id` only, we might have not matched the proper adjustment and vice versa.